### PR TITLE
昇腾和寒武纪相关代码退场 WITH_ASCEND_CL

### PR DESCRIPTION
**Pull request by Liyulingyue**
### PR types

Others 
### PR changes

Others 
### Describe

删除HCCL相关项

- https://github.com/PaddlePaddle/Paddle/issues/52319
- https://github.com/PaddlePaddle/Paddle/issues/52319#issuecomment-1490481708

### CODE REVIEW

**Comment by Liyulingyue** [comment]
~CI报错应该是和is_collective，麻烦Reviewer看一下怎么处理比较合适？~

上述问题应该是随机挂，re-run之后解决了

File: paddle/fluid/distributed/fleet_executor/test/CMakeLists.txt [hunk 1]
```diff
@@ -53,7 +53,7 @@ cc_test_old(
 
 if(WITH_DISTRIBUTE
    AND NOT WITH_PSLIB
-   AND NOT (WITH_ASCEND OR WITH_ASCEND_CL))
```
Comment (ronny1996) [inline paddle/fluid/distributed/fleet_executor/test/CMakeLists.txt:56]: WITH_ASCEND之后也可以删除

**Review by ronny1996 — DISMISSED** [review]
LGTM

File: paddle/fluid/distributed/fleet_executor/test/CMakeLists.txt (hunk 1 above)
Comment (Liyulingyue) [reply paddle/fluid/distributed/fleet_executor/test/CMakeLists.txt:56]: got it~

**Comment by luotao1** [comment]
需要解决下冲突

**Review by luotao1 — APPROVED** [review]

**Pull request merged**